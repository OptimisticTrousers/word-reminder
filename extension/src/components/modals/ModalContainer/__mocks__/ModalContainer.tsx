import { Props } from "../ModalContainer";

export function ModalContainer({ title, toggleModal, children }: Props) {
  return (
    <div>
      <h2>{title}</h2>
      <button aria-label="Close modal" onClick={toggleModal}>
        X
      </button>
      <div>{children}</div>
    </div>
  );
}
