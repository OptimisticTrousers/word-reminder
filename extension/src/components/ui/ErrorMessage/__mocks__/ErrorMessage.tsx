export function ErrorMessage({ message }: { message: string }) {
  return <div data-testid="error-message">{message}</div>;
}
