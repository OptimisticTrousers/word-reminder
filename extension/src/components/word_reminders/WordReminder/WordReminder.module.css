.word-reminder {
  background-color: var(--card-bkg-color);
  padding: 1rem;
  border-radius: 8px;
  border: 1px solid var(--border-color);
  margin-bottom: 0.75rem;
}

.word-reminder__buttons {
  display: flex;
  gap: 0.75rem;
  margin-top: 0.75rem;
}

.word-reminder__button {
  flex: 1;
}
