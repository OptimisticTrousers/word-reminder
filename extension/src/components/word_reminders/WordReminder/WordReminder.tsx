import {
  WordReminder as IWordReminder,
  Detail,
  UserWord as UserWordI,
} from "common";
import cronstrue from "cronstrue";
import { useState } from "react";
import CSSModules from "react-css-modules";

import { DeleteWordReminderModal } from "../../modals/DeleteWordReminderModal";
import { UpdateWordReminderModal } from "../../modals/UpdateWordReminderModal";
import { UserWord } from "../../words/UserWord";
import styles from "./WordReminder.module.css";
import { Link } from "react-router-dom";

export interface Props {
  wordReminder: IWordReminder & {
    user_words: (UserWordI & { details: Detail[] })[];
  };
}

export const WordReminder = CSSModules(
  function ({ wordReminder }: Props) {
    const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
    const [isUpdateModalOpen, setIsUpdateModalOpen] = useState(false);

    function toggleUpdateModal() {
      setIsUpdateModalOpen((prevIsUpdateModalOpen) => !prevIsUpdateModalOpen);
    }

    function toggleDeleteModal() {
      setIsDeleteModalOpen((prevIsDeleteModalOpen) => !prevIsDeleteModalOpen);
    }

    const reminder = cronstrue.toString(wordReminder.reminder, {
      throwExceptionOnParseError: false,
    });

    return (
      <>
        <div styleName="word-reminder">
          <div styleName="word-reminder__id">ID: {wordReminder.id}</div>
          <Link
            styleName="word-reminder__link"
            to={`/wordReminders/${wordReminder.id}`}
          >
            Word Reminder Details
          </Link>
          <div styleName="word-reminder__reminder">
            Reminder: {reminder} (UTC)
          </div>
          <div styleName="word-reminder__is-active">
            Active (whether the word reminder will actively remind you of the
            words in it): {wordReminder.is_active ? "Yes" : "No"}
          </div>
          <div styleName="word-reminder__has-reminder-onload">
            Has Reminder Onload (reminds you of these words when you open your
            browser): {wordReminder.has_reminder_onload ? "Yes" : "No"}
          </div>
          <time styleName="word-reminder__finish">
            This word reminder will become inactive on{" "}
            {wordReminder.finish.toLocaleString()}
          </time>
          <div styleName="word-reminder__created_at">
            This word reminder was created on{" "}
            {wordReminder.created_at.toLocaleString()}
          </div>
          <div styleName="word-reminder__updated_at">
            This word reminder was updated on{" "}
            {wordReminder.updated_at.toLocaleString()}
          </div>
          <div styleName="word-reminder__buttons">
            <button
              styleName="word-reminder__button"
              onClick={toggleUpdateModal}
            >
              Update
            </button>
            <button
              styleName="word-reminder__button"
              onClick={toggleDeleteModal}
            >
              Delete
            </button>
          </div>
          <ul styleName="word-reminder__user-words">
            {wordReminder.user_words.map(
              (user_word: UserWordI & { details: Detail[] }, index: number) => {
                return <UserWord key={index} {...user_word} />;
              }
            )}
          </ul>
        </div>
        {isUpdateModalOpen && (
          <UpdateWordReminderModal
            wordReminder={wordReminder}
            toggleModal={toggleUpdateModal}
          />
        )}
        {isDeleteModalOpen && (
          <DeleteWordReminderModal
            wordReminderId={String(wordReminder.id)}
            toggleModal={toggleDeleteModal}
          />
        )}
      </>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "ignore" }
);
