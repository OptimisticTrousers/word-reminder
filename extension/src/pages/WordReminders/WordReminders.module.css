.words-by-durations {
  height: 100%;
}
.words-by-durations__title {
  text-align: center;
}
