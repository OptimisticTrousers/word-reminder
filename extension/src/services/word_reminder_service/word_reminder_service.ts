import { AutoWordReminderParams, ManualWordReminderParams } from "common";

import { service, Params } from "../service";

export const wordReminderService = (function (service) {
  const { get, post, put, remove, VITE_API_DOMAIN } = service;

  function getWordReminderList(userId: string, params: Params) {
    return get({
      url: `${VITE_API_DOMAIN}/users/${userId}/wordReminders`,
      params,
      options: {
        credentials: "include",
      },
    });
  }

  function getWordReminder(userId: string, wordReminderId: string) {
    return get({
      url: `${VITE_API_DOMAIN}/users/${userId}/wordReminders/${wordReminderId}`,
      options: {
        credentials: "include",
      },
    });
  }

  function createWordReminder({
    userId,
    body,
  }: {
    userId: string;
    body: ManualWordReminderParams | AutoWordReminderParams;
  }) {
    return post({
      url: `${VITE_API_DOMAIN}/users/${userId}/wordReminders`,
      options: { body: JSON.stringify(body), credentials: "include" },
    });
  }

  function deleteWordReminders(userId: string) {
    return remove({
      url: `${VITE_API_DOMAIN}/users/${userId}/wordReminders`,
      options: { credentials: "include" },
    });
  }

  function updateWordReminder({
    userId,
    wordReminderId,
    body,
  }: {
    userId: string;
    wordReminderId: string;
    body: Omit<ManualWordReminderParams, "auto">;
  }) {
    return put({
      url: `${VITE_API_DOMAIN}/users/${userId}/wordReminders/${wordReminderId}`,
      options: { body: JSON.stringify(body), credentials: "include" },
    });
  }

  function deleteWordReminder({
    userId,
    wordReminderId,
  }: {
    userId: string;
    wordReminderId: string;
  }) {
    return remove({
      url: `${VITE_API_DOMAIN}/users/${userId}/wordReminders/${wordReminderId}`,
      options: { credentials: "include" },
    });
  }

  return {
    createWordReminder,
    deleteWordReminder,
    deleteWordReminders,
    getWordReminder,
    getWordReminderList,
    updateWordReminder,
  };
})(service);
