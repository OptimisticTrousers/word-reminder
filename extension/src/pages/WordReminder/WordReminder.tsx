import CSSModules from "react-css-modules";

import { WordReminder as WordReminderItem } from "../../components/word_reminders/WordReminder";
import styles from "./WordReminder.module.css";
import { useQuery } from "@tanstack/react-query";
import { useOutletContext, useParams } from "react-router-dom";
import { wordReminderService } from "../../services/word_reminder_service";
import { User } from "common";
import { Loading } from "../../components/ui/Loading";
import { Error500 } from "../Error500";

export const WordReminder = CSSModules(
  function () {
    const { user }: { user: User } = useOutletContext();
    const userId = String(user.id);
    const { wordReminderId } = useParams();
    const { data, failureReason, isLoading } = useQuery({
      queryKey: ["wordReminders", wordReminderId],
      queryFn: () => {
        return wordReminderService.getWordReminder({
          userId,
          wordReminderId: String(wordReminderId),
        });
      },
    });

    if (failureReason) {
      return <Error500 message={failureReason.message} />;
    }

    if (isLoading) {
      return <Loading />;
    }

    return <WordReminderItem wordReminder={data?.json.wordReminder} />;
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "log" }
);
