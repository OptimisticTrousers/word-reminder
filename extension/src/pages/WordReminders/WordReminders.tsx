import {
  User,
  WordReminder as IWordReminder,
  Detail,
  Column,
  UserWord,
} from "common";
import CSSModules from "react-css-modules";
import { useOutletContext, useSearchParams } from "react-router-dom";
import { keepPreviousData, useQuery } from "@tanstack/react-query";

import { CreateWordReminder } from "../../components/word_reminders/CreateWordReminder";
import { PaginatedList } from "../../components/ui/PaginatedList";
import { WordReminder } from "../../components/word_reminders/WordReminder";
import { wordReminderService } from "../../services/word_reminder_service/word_reminder_service";
import styles from "./WordReminders.module.css";
import { AutoCreateWordReminder } from "../../components/word_reminders/AutoCreateWordReminder";
import { ChangeEvent, FormEvent } from "react";

export const WordReminders = CSSModules(
  function () {
    const { user }: { user: User } = useOutletContext();
    const userId = String(user.id);
    const [searchParams, setSearchParams] = useSearchParams({
      page: "1",
      limit: PAGINATION_LIMIT,
      column: Column.CREATED_AT,
      direction: "-1",
    });
    const searchParamsObject = Object.fromEntries(searchParams);
    const { data, error, isLoading, isSuccess } = useQuery({
      queryKey: ["wordReminders", searchParamsObject],
      placeholderData: keepPreviousData,
      queryFn: () => {
        return wordReminderService.getWordReminderList({
          userId,
          params: searchParamsObject,
        });
      },
    });

    function handleQuery(event: FormEvent<HTMLFormElement>) {
      event.preventDefault();
      const formData = new FormData(event.currentTarget);
      const newSearchParams = new URLSearchParams({
        column:
          ((formData.get(Column.CREATED_AT) as string) && Column.CREATED_AT) ||
          ((formData.get(Column.UPDATED_AT) as string) && Column.UPDATED_AT) ||
          "",
        direction:
          (formData.get(Column.CREATED_AT) as string) ||
          (formData.get(Column.UPDATED_AT) as string) ||
          "",
      });
      const combined = new URLSearchParams([
        ...searchParams,
        ...newSearchParams,
      ]);
      setSearchParams(combined);
    }

    function handleSelectChange(event: ChangeEvent<HTMLSelectElement>) {
      const selectElement = event.target;
      const selectedOption = selectElement.selectedOptions[0];
      const optgroup = selectedOption.parentElement as HTMLOptGroupElement;
      const label = optgroup.label;

      let name = "";
      switch (label) {
        case "Created At":
          name = "created_at";
          break;
        case "Updated At":
          name = "updated_at";
          break;
      }

      selectElement.name = name;
    }

    function setPagination(limit: number, page: number) {
      setSearchParams((prevSearchParams) => {
        return { ...prevSearchParams, limit, page };
      });
    }

    const json = data?.json;

    return (
      <div styleName="words-reminder">
        <AutoCreateWordReminder />
        <CreateWordReminder />
        <form styleName="word-reminder__form" onSubmit={handleQuery}>
          <fieldset styleName="word-reminder__sort sort">
            <label styleName="sort__label" htmlFor="sort">
              Sort by:
            </label>
            <select
              id="sort"
              styleName="sort__select"
              onChange={handleSelectChange}
            >
              <option styleName="sort__option" value="">
                Featured
              </option>
              <optgroup label="Created At">
                <option styleName="sort__option" value="1">
                  Oldest Created
                </option>
                <option styleName="sort__option" value="-1">
                  Newest Created
                </option>
              </optgroup>
              <optgroup label="Updated At">
                <option styleName="sort__option" value="1">
                  Oldest Updated
                </option>
                <option styleName="sort__option" value="-1">
                  Newest Updated
                </option>
              </optgroup>
            </select>
          </fieldset>
          <button styleName="word-reminder__button" type="submit">
            Filter
          </button>
        </form>
        <PaginatedList
          name="Word Reminders"
          totalRows={json && json.totalRows}
          list={
            json &&
            json.wordReminders.map(function (
              wordReminder: IWordReminder & {
                user_words: (UserWord & { details: Detail[] })[];
              }
            ) {
              return (
                <WordReminder
                  key={wordReminder.id}
                  wordReminder={wordReminder}
                />
              );
            })
          }
          isLoading={isLoading}
          isSuccess={isSuccess}
          error={error}
          previous={json && json.previous}
          next={json && json.next}
          setPagination={setPagination}
        />
      </div>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "log" }
);

const PAGINATION_LIMIT = "10";
