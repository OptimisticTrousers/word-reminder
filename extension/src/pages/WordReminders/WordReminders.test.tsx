import { render, screen } from "@testing-library/react";

import { WordReminder } from "../../components/word_reminders/WordReminder/__mocks__/WordReminder";
import { wordReminderService } from "../../services/word_reminder_service";
import { WordReminders } from "./WordReminders";
import userEvent from "@testing-library/user-event";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { createRoutesStub, Outlet } from "react-router-dom";

vi.mock("../../components/word_reminders/AutoCreateWordReminder", () => {
  return {
    AutoCreateWordReminder: function () {
      return <div data-testid="auto-create-word-reminder"></div>;
    },
  };
});

vi.mock("../../components/word_reminders/CreateWordReminder", () => {
  return {
    CreateWordReminder: function () {
      return <div data-testid="create-word-reminder"></div>;
    },
  };
});

vi.mock("../../components/ui/PaginatedList/PaginatedList");

vi.mock("../../components/word_reminders/WordReminder/WordReminder");

describe("WordReminders component", () => {
  const testUser = {
    id: 1,
  };

  const word1 = {
    id: 1,
    details: [
      {
        word: "hello",
        phonetics: [
          {
            text: "hɛˈləʊ",
          },
        ],
        meanings: [
          {
            partOfSpeech: "exclamation",
            definitions: [
              {
                definition:
                  "used as a greeting or to begin a phone conversation.",
                example: "hello there, Katie!",
                synonyms: [],
                antonyms: [],
              },
            ],
          },
        ],
      },
    ],
    created_at: new Date(),
  };

  const word2 = {
    id: 2,
    details: [
      {
        word: "clemency",
        meanings: [
          {
            partOfSpeech: "noun",
            definitions: [{ definition: "Mercy; lenience." }],
          },
        ],
        phonetics: [],
      },
    ],
    created_at: new Date(),
  };

  const userWord1 = {
    id: 1,
    user_id: testUser.id,
    word_id: word1.id,
    details: word1.details,
    learned: false,
    created_at: new Date(),
    updated_at: new Date(),
  };

  const userWord2 = {
    id: 2,
    user_id: testUser.id,
    word_id: word2.id,
    details: word2.details,
    learned: false,
    created_at: new Date(),
    updated_at: new Date(),
  };

  const wordReminder1 = {
    id: 1,
    has_reminder_onload: true,
    is_active: true,
    user_id: testUser.id,
    reminder: "* * * * *",
    finish: new Date(Date.now() + 1000),
    user_words: [userWord1, userWord2],
    created_at: new Date(),
    updated_at: new Date(),
  };

  const wordReminder2 = {
    id: 2,
    has_reminder_onload: true,
    is_active: true,
    user_id: testUser.id,
    reminder: "* * * * *",
    finish: new Date(Date.now() + 1000),
    user_words: [userWord1, userWord2],
    created_at: new Date(),
    updated_at: new Date(),
  };

  const wordReminders = [wordReminder1, wordReminder2];

  const status = 200;

  const PAGINATION_LIMIT = "10";

  const searchParams = new URLSearchParams();

  function setup() {
    const queryClient = new QueryClient();
    const Stub = createRoutesStub([
      {
        path: "/",
        Component: function () {
          return <Outlet context={{ user: testUser }} />;
        },
        children: [
          {
            path: "/",
            Component: function () {
              return (
                <QueryClientProvider client={queryClient}>
                  <WordReminders />
                </QueryClientProvider>
              );
            },
          },
        ],
      },
    ]);

    return {
      user: userEvent.setup(),
      ...render(<Stub initialEntries={["/"]} />),
    };
  }

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("renders the form that allows user to filter the word reminders and paginated list", async () => {
    const mockGetWordReminderList = vi
      .spyOn(wordReminderService, "getWordReminderList")
      .mockImplementation(async () => {
        return {
          json: {
            wordReminders,
            next: {
              page: 3,
              limit: PAGINATION_LIMIT,
            },
            previous: { page: 1, limit: PAGINATION_LIMIT },
            totalRows: 2,
          },
          status,
        };
      });
    const { asFragment } = setup();

    const paginatedListProps = {
      name: "Word Reminders",
      totalRows: wordReminders.length,
      list: [
        <WordReminder
          searchParams={searchParams}
          wordReminder={wordReminder1}
        />,
        <WordReminder
          searchParams={searchParams}
          wordReminder={wordReminder2}
        />,
      ],
      isLoading: false,
      isSuccess: true,
      error: null,
      previous: { page: 1, limit: PAGINATION_LIMIT },
      next: { page: 3, limit: PAGINATION_LIMIT },
    };
    const autoCreateWordReminder = screen.getByTestId(
      "auto-create-word-reminder"
    );
    const createWordReminder = screen.getByTestId("create-word-reminder");
    const paginatedList = await screen.findByTestId("paginated-list");
    expect(paginatedList).toBeInTheDocument();
    expect(paginatedList).toHaveTextContent(String(paginatedListProps));
    expect(autoCreateWordReminder).toBeInTheDocument();
    expect(createWordReminder).toBeInTheDocument();
    expect(mockGetWordReminderList).toHaveBeenCalledTimes(1);
    expect(mockGetWordReminderList).toHaveBeenCalledWith({
      userId: String(testUser.id),
      params: {
        page: "1",
        limit: "10",
        column: "",
        direction: "",
      },
    });
    expect(asFragment()).toMatchSnapshot();
  });

  describe("sort by options", () => {
    it("allows the user to sort by featured", async () => {
      const mockGetWordReminderList = vi
        .spyOn(wordReminderService, "getWordReminderList")
        .mockImplementation(async () => {
          return {
            json: {
              wordReminders,
              next: {
                page: 3,
                limit: PAGINATION_LIMIT,
              },
              previous: { page: 1, limit: PAGINATION_LIMIT },
            },
            status,
          };
        });
      const { user } = setup();

      const select = screen.getByRole("combobox");
      const button = screen.getByRole("button", { name: "Filter" });
      await user.selectOptions(select, ["Featured"]);
      await user.click(button);

      expect(mockGetWordReminderList).toHaveBeenCalledTimes(1);
      expect(mockGetWordReminderList).toHaveBeenCalledWith({
        userId: String(testUser.id),
        params: Object.fromEntries(
          new URLSearchParams({
            page: "1",
            limit: PAGINATION_LIMIT,
            column: "",
            direction: "",
          })
        ),
      });
    });

    it("allows the user to sort by newest", async () => {
      const mockGetWordReminderList = vi
        .spyOn(wordReminderService, "getWordReminderList")
        .mockImplementation(async () => {
          return {
            json: {
              wordReminders,
              next: {
                page: 3,
                limit: PAGINATION_LIMIT,
              },
              previous: { page: 1, limit: PAGINATION_LIMIT },
            },
            status,
          };
        });
      const { user } = setup();

      const select = screen.getByRole("combobox");
      const button = screen.getByRole("button", { name: "Filter" });
      await user.selectOptions(select, ["Newest"]);
      await user.click(button);

      expect(mockGetWordReminderList).toHaveBeenCalledTimes(2);
      expect(mockGetWordReminderList).toHaveBeenCalledWith({
        userId: String(testUser.id),
        params: Object.fromEntries(
          new URLSearchParams({
            page: "1",
            limit: PAGINATION_LIMIT,
            column: "",
            direction: "",
          })
        ),
      });
      expect(mockGetWordReminderList).toHaveBeenCalledWith({
        userId: String(testUser.id),
        params: Object.fromEntries(
          new URLSearchParams({
            page: "1",
            limit: PAGINATION_LIMIT,
            column: "created_at",
            direction: "1",
          })
        ),
      });
    });

    it("allows the user to sort by oldest", async () => {
      const mockGetWordReminderList = vi
        .spyOn(wordReminderService, "getWordReminderList")
        .mockImplementation(async () => {
          return {
            json: {
              wordReminders,
              next: {
                page: 3,
                limit: PAGINATION_LIMIT,
              },
              previous: { page: 1, limit: PAGINATION_LIMIT },
            },
            status,
          };
        });
      const { user } = setup();

      const select = screen.getByRole("combobox");
      const button = screen.getByRole("button", { name: "Filter" });
      await user.selectOptions(select, ["Oldest"]);
      await user.click(button);

      expect(mockGetWordReminderList).toHaveBeenCalledTimes(2);
      expect(mockGetWordReminderList).toHaveBeenCalledWith({
        userId: String(testUser.id),
        params: Object.fromEntries(
          new URLSearchParams({
            page: "1",
            limit: PAGINATION_LIMIT,
            column: "",
            direction: "",
          })
        ),
      });
      expect(mockGetWordReminderList).toHaveBeenCalledWith({
        userId: String(testUser.id),
        params: Object.fromEntries(
          new URLSearchParams({
            page: "1",
            limit: PAGINATION_LIMIT,
            column: "created_at",
            direction: "-1",
          })
        ),
      });
    });
  });
});
