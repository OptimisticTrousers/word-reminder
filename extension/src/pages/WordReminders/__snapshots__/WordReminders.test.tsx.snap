// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WordReminders component > renders the form that allows user to filter the word reminders and paginated list 1`] = `
<DocumentFragment>
  <div
    class="_words-reminder_928901"
  >
    <div
      data-testid="auto-create-word-reminder"
    />
    <div
      data-testid="create-word-reminder"
    />
    <form
      action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')"
      class="_word-reminder__form_928901"
    >
      <div
        class="_sort_7b9046"
      >
        <label
          class="_sort__label_7b9046"
        >
          Sort by:
          <select
            class="_sort__select_7b9046"
            name="created_at"
            required=""
          >
            <option
              class="_sort__option_7b9046"
              value=""
            >
              Featured
            </option>
            <option
              class="_sort__option_7b9046"
              value="1"
            >
              Newest
            </option>
            <option
              class="_sort__option_7b9046"
              value="-1"
            >
              Oldest
            </option>
          </select>
        </label>
      </div>
      <button
        class="_word-reminder__button_928901"
        type="submit"
      >
        Filter
      </button>
    </form>
    <div
      data-testid="paginated-list"
    >
      [object Object]
    </div>
  </div>
</DocumentFragment>
`;
